src/admittance_model.jl

```diff
@@ -130,127 +139,112 @@ function cascade(ams::AbstractVector{U}) where {T, U <: AdmittanceModel{T}}
     ports = vcat([get_ports(am) for am in ams]...)
     return partial_copy(ams[1], Y=Y, P=P, Q=Q, ports=ports)
 end
-
-cascade(ams::Vararg{U}) where {T, U <: AdmittanceModel{T}} = cascade(collect(ams))
+cascade(ams::AdmittanceModel...) = cascade(collect(ams))
 
 """
-    ports_to_indices(am::AdmittanceModel{T}, ports::AbstractVector{T}) where T
-    ports_to_indices(am::AdmittanceModel{T}, ports::Vararg{T}) where T
+    ports_to_indices(am::AdmittanceModel, ports::AbstractVector)
+    ports_to_indices(am::AdmittanceModel, ports...)
 
 Find the indices corresponding to given ports.
 """
-function ports_to_indices(am::AdmittanceModel{T}, ports::AbstractVector{T}) where T
+function ports_to_indices(am::AdmittanceModel, ports::AbstractVector)
```

Comment:
does this remove the restriction of Abstractvector and AdmittanceModel have to be the same type? would it matter if they are not?

Reply:
nevermind, just read your comment in PR.

Reply:
This broadens the types of arguments that can be passed to `ports_to_indices`. To be honest it probably doesn't matter much either way here, but the way it is written now, you couldn't pass a vector of `SubString` if you have an `AdmittanceModel{String}` (admittance model with `String` port names). I think the restriction is more likely to cause unexpected frustration than not.

My approach to argument typing is basically to be as generic as possible consistent with hinting to the user what you have in mind. Making the argument types more specific than that often just makes your code less able to handle generic cases. A good use case for making argument types very specific is when you have implementations of a function that differ depending on argument types (e.g. there are many methods of `+` for different numeric types, like `Float64` or `Int`).